tests/unit-tests/Commands/RunCommandTest.php

```diff
@@ -53,7 +54,7 @@ public function can_proxy_artisan_commands()
      */
     public function proxies_exceptions()
     {
-        $this->expectException(\Exception::class);
+        $this->expectException(Error::class);
```

Comment:
Why the change from Exception to Error?

Reply:
Laravel 6 throws an error, instead of an exception, here. Feel free to run the test with it set to exception, perhaps I misinterpreted it, and the error is due to something else?

Reply:
Here's the output:
```
1) Hyn\Tenancy\Tests\Commands\RunCommandTest::proxies_exceptions
Failed asserting that exception of type "Error" matches expected exception "Exception". Message was: "Call to undefined method Exception::run()" at
/Users/mike/Developer/Sites/multi-tenant/vendor/laravel/framework/src/Illuminate/Console/Command.php:231
/Users/mike/Developer/Sites/multi-tenant/vendor/laravel/framework/src/Illuminate/Console/Command.php:204
/Users/mike/Developer/Sites/multi-tenant/src/Commands/RunCommand.php:82
/Users/mike/Developer/Sites/multi-tenant/vendor/laravel/framework/src/Illuminate/Database/Concerns/BuildsQueries.php:39
/Users/mike/Developer/Sites/multi-tenant/src/Commands/RunCommand.php:85
/Users/mike/Developer/Sites/multi-tenant/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:32
/Users/mike/Developer/Sites/multi-tenant/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:90
/Users/mike/Developer/Sites/multi-tenant/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:34
/Users/mike/Developer/Sites/multi-tenant/vendor/laravel/framework/src/Illuminate/Container/Container.php:591
/Users/mike/Developer/Sites/multi-tenant/vendor/laravel/framework/src/Illuminate/Console/Command.php:192
/Users/mike/Developer/Sites/multi-tenant/vendor/symfony/console/Command/Command.php:255
/Users/mike/Developer/Sites/multi-tenant/vendor/laravel/framework/src/Illuminate/Console/Command.php:179
/Users/mike/Developer/Sites/multi-tenant/vendor/symfony/console/Application.php:915
/Users/mike/Developer/Sites/multi-tenant/vendor/symfony/console/Application.php:272
/Users/mike/Developer/Sites/multi-tenant/vendor/symfony/console/Application.php:148
/Users/mike/Developer/Sites/multi-tenant/vendor/laravel/framework/src/Illuminate/Console/Application.php:90
/Users/mike/Developer/Sites/multi-tenant/vendor/laravel/framework/src/Illuminate/Console/Application.php:182
/Users/mike/Developer/Sites/multi-tenant/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php:273
/Users/mike/Developer/Sites/multi-tenant/vendor/laravel/framework/src/Illuminate/Foundation/Testing/Concerns/InteractsWithConsole.php:43
/Users/mike/Developer/Sites/multi-tenant/tests/unit-tests/Commands/RunCommandTest.php:64
```

Reply:
That looks like it's throwing an `Error` because you're calling a method that doesn't exist on that class.  In this case, it looks like `$this->resolveCommand()` is returning an `Exception` instead of a `Command`.  At quick glance, it seems the only way that can happen is if the object passed in is already an `Exception`

Reply:
Can you attempt to log the contents of `$this->argument('run')` right before that call? https://github.com/tenancy/multi-tenant/blob/5.x/src/Commands/RunCommand.php#L81

If it looks good, step in and repeat.  You'll probably have to try to reproduce the failure locally first, assuming you haven't already

Reply:
Output of `dump($this->argument('run'), $options->toArray());` is:
```
"exception"
[]
```

Reply:
I'm not sure what you are expecting back. Is `exception` a valid method? I would assume not? If not, the response seems appropriate to me.

Reply:
In theory calling `run` with `run = exception` is trying to execute the `exception` command defined in RunCommandTest on lines 30-32-ish.

Reply:
`dump($this->resolveCommand($command));` is what returns the exception:
```
Exception {#1354
  #message: ""
  #code: 0
  #file: "./vendor/laravel/framework/src/Illuminate/Console/Command.php"
  #line: 248
  trace: {
    ./vendor/laravel/framework/src/Illuminate/Console/Command.php:248 { …}
    ./vendor/laravel/framework/src/Illuminate/Console/Command.php:230 { …}
    ./vendor/laravel/framework/src/Illuminate/Console/Command.php:204 { …}
    ./src/Commands/RunCommand.php:82 {
      ›     $this->argument('run'),
      ›     $options->toArray()
      › );
      arguments: {
        $command: "exception"
        $arguments: []
      }
    }
    ./vendor/laravel/framework/src/Illuminate/Database/Concerns/BuildsQueries.php:39 { …}
    ./src/Commands/RunCommand.php:85 { …}
    Hyn\Tenancy\Commands\RunCommand->handle() {}
    ./vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:32 { …}
    ./vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:90 { …}
    ./vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:34 { …}
    ./vendor/laravel/framework/src/Illuminate/Container/Container.php:591 { …}
    ./vendor/laravel/framework/src/Illuminate/Console/Command.php:192 { …}
    ./vendor/symfony/console/Command/Command.php:255 { …}
    ./vendor/laravel/framework/src/Illuminate/Console/Command.php:179 { …}
    ./vendor/symfony/console/Application.php:915 { …}
    ./vendor/symfony/console/Application.php:272 { …}
    ./vendor/symfony/console/Application.php:148 { …}
    ./vendor/laravel/framework/src/Illuminate/Console/Application.php:90 { …}
    ./vendor/laravel/framework/src/Illuminate/Console/Application.php:182 { …}
    ./vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php:273 { …}
    ./vendor/laravel/framework/src/Illuminate/Foundation/Testing/Concerns/InteractsWithConsole.php:43 { …}
    ./tests/unit-tests/Commands/RunCommandTest.php:63 { …}
    ./vendor/phpunit/phpunit/src/Framework/TestCase.php:1327 { …}
    ./vendor/phpunit/phpunit/src/Framework/TestCase.php:947 { …}
    ./vendor/phpunit/phpunit/src/Framework/TestResult.php:691 { …}
    ./vendor/phpunit/phpunit/src/Framework/TestCase.php:679 { …}
    ./vendor/phpunit/phpunit/src/Framework/TestSuite.php:568 { …}
    ./vendor/phpunit/phpunit/src/Framework/TestSuite.php:568 { …}
    ./vendor/phpunit/phpunit/src/Framework/TestSuite.php:568 { …}
    ./vendor/phpunit/phpunit/src/TextUI/TestRunner.php:616 { …}
    ./vendor/phpunit/phpunit/src/TextUI/Command.php:201 { …}
    ./vendor/phpunit/phpunit/src/TextUI/Command.php:160 { …}
    ./vendor/phpunit/phpunit/phpunit:61 { …}
  }
}
```

Reply:
Thanks @drbyte that provided the info I needed. Implementing fix now.

Reply:
The issue is that it's passing in "exception", which [this code](https://github.com/laravel/framework/blob/6.x/src/Illuminate/Console/Command.php#L248) then tries to instantiate.  So you get an `\Exception` object which isn't a `Command` and therefore doesn't have a `->run(...)` method

Reply:
Hmm, I thought I had something, but it was a red herring. I'm coming up empty. The reason an error is being thrown is because it doesn't recognize the dynamically created command. It doesn't appear to exist.

Reply:
so the issue is that our test is running with `exception` as an "invalid" command name.  We should update the test to something like `commandThatDoesNotExist`

Reply:
@fletch3555 Bingo!

Reply:
My guess is that it worked before because the code was different: https://github.com/laravel/framework/blob/5.8/src/Illuminate/Console/Command.php#L213

Instead of instantiating the class directly, it tries to pull it from IOC, which I'm guessing threw an exception instead of blindly continuing

Reply:
Excellent!